central/role/datastore/datastore_impl.go

```diff
@@ -161,6 +169,14 @@ func (ds *dataStoreImpl) GetAllPermissionSets(ctx context.Context) ([]*storage.P
 	return permissionSets, nil
 }
 
+func (ds *dataStoreImpl) CountPermissionSets(ctx context.Context) (int, error) {
+	if ok, err := roleSAC.ReadAllowed(ctx); !ok || err != nil {
+		return 0, err
+	}
+
+	return ds.CountPermissionSets(ctx)
```

Comment:
Shouldn't it be `ds.permissionSetStorage.Count(ctx)` ?

Reply:
OMG returns